### PR TITLE
[SEC-102] Support retrieving secrets from SSM Parameter Store

### PR DESCRIPTION
**Pull request by bhatman**

### CODE REVIEW

File: clrenv/lazy_env.py [hunk 1]
```diff
@@ -6,14 +6,15 @@
 from itertools import chain, groupby
 import shlex
 import sys
-import types
```
Comment (bhatman) [inline clrenv/lazy_env.py:9]: not used

File: clrenv/lazy_env.py [hunk 2]
```diff
@@ -6,14 +6,15 @@
...
 
 from munch import Munch, munchify
 import yaml
 
 from .path import find_environment_path, find_user_environment_paths
 from functools import reduce
 
+import boto3
+SSM_CLIENT = boto3.client('ssm')
```
Comment (bhatman) [inline clrenv/lazy_env.py]: should probably lazy-load this

Comment (youssefm) [reply clrenv/lazy_env.py]: yeah definitely - you could memoize this inside a function to do that

File: clrenv/lazy_env.py [hunk 3]
```diff
@@ -162,19 +163,26 @@ def _apply_functions(d, recursive=False):
 
       ^function rest
 
-    Currently, the only function available is `keyfile', which attempts
-    to replace with a value from the currently loaded keyfile."""
+    Available functions:
+      ^keyfile: Looks up the given value in the current environment's clrypt keyfile.
+      ^secret: Looks up the given value in AWS Parameter store.
```
Comment (youssefm) [inline clrenv/lazy_env.py]: looks like you called it `^parameter` below not secret?

**Review by youssefm — COMMENTED** [review]
lazy load the client is important I think - we don't want to be making API calls if there are no `^parameter` s in the environment

File: clrenv/lazy_env.py (hunk 2 above)
Comment (bhatman) [reply clrenv/lazy_env.py]: done, thanks

**Review by youssefm — APPROVED** [review]
:shipit: 👍

**Pull request merged**